### PR TITLE
feat(Analytics): Support for Timeseries Aggregated Statistics

### PR DESCRIPTION
**Pull request by rslanka**
Adds support for doing aggregations over time-series aspect fields, and migrates the existing usage stats implementation on top this feature. The key changes include
- two new annotations @TimeseriesField and @TimeseriesFieldCollection to the PDL for defining the aggregatable fields
- necessary elastic index generation logic
- a generalized query end-point via the new analytics end-point.

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/linkedin/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable)

### CODE REVIEW

File: metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java [hunk 1]
```diff
@@ -44,9 +44,7 @@
 import lombok.extern.slf4j.Slf4j;
 
 import static com.linkedin.metadata.Constants.*;
-import static com.linkedin.metadata.entity.ebean.EbeanUtils.parseSystemMetadata;
-import static com.linkedin.metadata.entity.ebean.EbeanUtils.toAspectRecord;
-import static com.linkedin.metadata.entity.ebean.EbeanUtils.toJsonAspect;
+import static com.linkedin.metadata.entity.ebean.EbeanUtils.*;
```
Comment (dexter-mh-lee) [inline metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java:47]: Can we revert this file? Seems like the only change is on the comments. Let's just change that and make it clear what we are changing.

**Review by dexter-mh-lee — COMMENTED** [review]
If we have a string grouping and then date grouping, will it work? 

In general, we need more docs (Logic overview, API examples, etc.). Maybe add them in the next PR!

**Comment by rslanka** [comment]
> If we have a string grouping and then date grouping, will it work?
@dexter-mh-lee, Yes, it would work! The only limitation on the grouping buckets is that if a date grouping bucket is specified, it is allowed only on the `timestampMillis` field because of the way the index is built.

> In general, we need more docs (Logic overview, API examples, etc.). Maybe add them in the next PR!
Sure, will ship them in the next PR.

File: metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java (hunk 1 above)
Comment (rslanka) [reply metadata-io/src/main/java/com/linkedin/metadata/entity/ebean/EbeanEntityService.java:47]: The static imports change in this file is a result of running the linkedin code formatter on this file!

File: entity-registry/src/main/java/com/linkedin/metadata/models/EntitySpecBuilder.java [hunk 2]
```diff
@@ -314,4 +322,16 @@ private void validateKeyAspect(@Nonnull final AspectSpec keyAspect) {
...
   }
+
+  public enum AnnotationExtractionMode {
+    /**
+     * Extract all annotations types, the default.
+     */
+    DEFAULT,
+    /**
+     * Skip annotations on aspect record fields, only
+     * parse entity + aspect annotations.
```
Comment (jjoyce0510) [inline entity-registry/src/main/java/com/linkedin/metadata/models/EntitySpecBuilder.java:333]: Minor: might be useful to describe briefly why we need this mode

**Review by shirshanka — APPROVED** [review]
LGTM!

**Pull request merged**